static/app/components/searchSyntax/utils.tsx

```diff
@@ -241,3 +241,60 @@ export function isWithinToken(
 export function isOperator(value: string) {
   return allOperators.some(op => op === value);
 }
+
+function stringifyTokenFilter(token: TokenResult<Token.FILTER>) {
+  let stringifiedToken = '';
+
+  if (token.negated) {
+    stringifiedToken += '!';
+  }
+
+  stringifiedToken += stringifyToken(token.key);
+  stringifiedToken += ':';
+  stringifiedToken += token.operator;
+  stringifiedToken += stringifyToken(token.value);
+
+  return stringifiedToken;
+}
+
+export function stringifyToken(token: TokenResult<Token>) {
```

Comment:
This function allows us to modify the AST token object and generate the text output from our changes. It's similar to the search syntax renderer except that it outputs a string instead of JSX.